test/e2e/clusterctl_upgrade.go

```diff
@@ -691,6 +722,22 @@ func setupClusterctl(ctx context.Context, clusterctlBinaryURL, clusterctlConfigP
 	return clusterctlBinaryPath, clusterctlConfigPath
 }
 
+func getInfraMachinePool(ctx context.Context, managementClusterClient client.Client, mp expv1.MachinePool, workloadClusterNamespace, workloadClusterName string) *unstructured.Unstructured {
+	infraMachinePoolList := &unstructured.UnstructuredList{}
+	infraMachinePoolList.SetGroupVersionKind(mp.Spec.Template.Spec.InfrastructureRef.GroupVersionKind())
+	if err := managementClusterClient.List(
```

Comment:
Please let's make this a get call based on the InfrastructureRef

I think otherwise we can end up with the wrong InfraMachinePool here if there is more than 1 InfraMachinePool of a gvk for a cluster

Reply:
I think we should also return an error if we don't find the InfraMachinePool and not ignore it